runtime/ms_rest_azure/lib/ms_rest_azure/resource.rb

```diff
@@ -23,6 +23,15 @@ class Resource
     # @return [Hash{String => String}] the tags attached to resources (optional).
     attr_accessor :tags
 
+    # @return [String] the name of the resource group of the resource.
+    def resource_group
+      unless self.id.nil?
+        if self.id =~ /.+\/resourceGroups\/.+\/.+/

```

Comment:
I'd recommend below pattern as Group based matching, so that you don't have to split by `/` afterwards:

```
groups = self.id.match(/.*\/resourceGroups\/(.*)\/.*/)
groups.captures[0] if groups
```